src/htmlgen.cpp

```diff
@@ -1136,10 +1136,17 @@ void HtmlGenerator::startFile(const char *name,const char *,
   bool searchEngine = Config_getBool(SEARCHENGINE);
   if (searchEngine /*&& !generateTreeView*/)
   {
+    QCString htmlFileExtension=Config_getString(HTML_FILE_EXTENSION);
+    htmlFileExtension=htmlFileExtension.stripWhiteSpace();
+    if (htmlFileExtension.isEmpty())
+    {
+      htmlFileExtension = ".html";
+    }
+
     t << "<script type=\"text/javascript\">\n";
 		t << "/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */\n";
-		t << "var searchBox = new SearchBox(\"searchBox\", \""
-      << m_relPath<< "search\",false,'" << theTranslator->trSearch() << "');\n";
+		t << "var searchBox = new SearchBoxExt(\"searchBox\", \""
+      << m_relPath<< "search\",false,'" << theTranslator->trSearch() << "','" << htmlFileExtension << "');\n";
```

Comment:
- Why create a new function and not just adjust the current function.
- Why only correct 2 out of 3 `SearchBox` (missing `void HtmlGenerator::writeExternalSearchPage()` ?

Reply:
Given my lack of knowledge of the project and the impossibility of testing external research, I wanted to be conservative.

Now I also applied the fix to void HtmlGenerator::writeExternalSearchPage() and restored the single function in search.js